jetbrains-core/src/software/aws/toolkits/jetbrains/services/schemas/code/SchemaCodeDownloader.kt

```diff
@@ -66,20 +68,27 @@ class SchemaCodeDownloader(
     }
 
     companion object {
-        fun create(project: Project): SchemaCodeDownloader = SchemaCodeDownloader(
-            CodeGenerator(project.awsClient()),
-            CodeGenerationStatusPoller(project.awsClient()),
-            CodeDownloader(project.awsClient()),
-            CodeExtractor(),
-            ProgressUpdater()
-        )
+        fun create(project: Project): SchemaCodeDownloader {
+            val connectionSettings = AwsConnectionManager.getInstance(project).connectionSettings()
+                ?: TODO()
```

Comment:
Fill todo?

Reply:
throws exception, dont use this with an invalid connection

Reply:
Yeah but why not throw real exception `IllegalStateException`?

Reply:
It does now